snapcraft/internal/project_loader/_config.py

```diff
@@ -209,6 +210,15 @@ def __init__(self, project: project.Project) -> None:
         snapcraft_yaml = self._expand_filesets(snapcraft_yaml)
 
         self.data = self._expand_env(snapcraft_yaml)
+
+        self.data["architectures"] = _process_architectures(
+            self.data.get("architectures"), project.deb_arch
+        )
+
+        # TODO: convergence of project_loader and config.
+        snap_meta = Snap.from_dict(self.data)
```

Comment:
This should probably be the holder of snap_meta, we should work towards this goal.
Can you make the on in Project private?

Reply:
Let's discuss this more

Reply:
Made private, and consolidated to the instance in Project() being overriden by Config.